core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -396,6 +396,10 @@ public boolean sync() throws AccessControlException, InvalidPathException {
               elapsedTime.toMillis(), mRootScheme);
     }
     boolean success = syncPathCount > 0;
+    if (ServerConfiguration.getBoolean(PropertyKey.MASTER_METADATA_SYNC_REPORT_FAILURE)) {
+      // There should not be any failed or outstanding jobs
+      success = (failedSyncPathCount == 0) && mSyncPathJobs.isEmpty() && mPendingPaths.isEmpty();
+    }
     if (success) {
       // update the sync path cache for the root of the sync
```

Comment:
What is the side effect now that we skip the sync path cache update on partial syncs?

Reply:
if we did a `ls -R /path` and failed, we should not update the sync path cache. Otherwise, any path under /path will be considered synced when they are in fact not in sync.

Reply:
Is there some way we can have a unittest which confirms this behavior?

Reply:
added

Reply:
added the test which does three things:
1. simulates an interrupted syncmetadata
2. check to make sure it leaves no incomplete files
3. check to make sure the files are not considered synced by the file master